src/assignment/views/AssignmentEdit.tsx

```diff
@@ -198,6 +199,11 @@ const AssignmentEdit: FunctionComponent<DefaultSecureRouteProps<{ id: string }>>
 			return;
 		}
 
+		if (assignment.deadline_at && isPast(new Date(assignment.deadline_at))) {
+			ToastService.danger(t('De deadline mag niet in het verleden liggen.'));
+			return;
+		}
```

Comment:
I'd still argue this is best added in the service

Reply:
isn't it better to warn the user about validation errors, before showing any confirmation dialogs?